oio/container/backup.py

```diff
@@ -506,24 +621,38 @@ def parse_reg_type(self, hdrs, account, container):
             kwargs['mime_type'] = hdrs['mime_type']
             del hdrs['mime_type']
 
-        data = LimitedStream(self.req.stream, self.inf.size)
+        data = LimitedStream(self.req.stream, self.inf.size,
+                             entry=self.cur_state.get('entry'),
+                             offset=self.cur_state.get('offset'))
         try:
             _, size, _ = self.proxy.object_create(
                 account, container, obj_name=self.inf.name, append=self.append,
                 file_or_path=data, **kwargs)
-        except Exception:
+        except (Exception, exc.SourceReadError) as ex:
             # no data was written if error occurs during object_create
             # we just have to update our state_machine offset regarding
             # current object
-            if self.mode == self.MODE_FULL:
+            if self.cur_state.get('manifest') is None:
                 raise
+
+            entry = None
             for entry in self.cur_state['manifest']:
                 if entry['name'] == self.inf.name:
                     break
             else:  # it should not happen
                 raise BadRequest("Invalid internal state")
+
+            # since an error has occured, we have to reset
+            # current offset to start of current chunk
+            # and remove stored checksum if set
+
+            if isinstance(ex, exc.SourceReadError):
+                self.logger.error("Invalid checksum detected for %s",
+                                  self.inf.name)
+                raise BadRequest("Checksum error %s" % self.inf.name)
```

Comment:
How do we know this error comes from a bad checksum?

Reply:
Hum, I'm agree that it is a shortcut... I will add a flag in LimitedStream and check here to remove ambiguity regarding other possible errors